### ISSUE TITLE
Activity Monitoring

### ISSUE THREAD

**Issue by HesselM** [HesselM/IN4254-SmartPhoneSensing#3]
1. get accel data.
   - Select action X, perform action X and do Y measurements of accel over X time.
   - label data.
   - store data??? (will be a lot!).
     - table 'measurement-time'.
     - auto increment ID [int]
     - run ID [int]
     - timestamp [int - UTC microseconds??]
     - action [int: 0=walking (not in queue/exit), 1=queueing (start/not moving), 2=moving (moving in queue)]
     - x-activity [float]
     - y-activity [float]
     - z-activity [float]
2. Learn (apply filters/classification)
   - transform measurements from time to frequency domain (FFT - Z bins ??)
   - apply gaussian filter/alpha filtering??
   - store results:
     - table 'classification'
     - auto increment ID [int]
     - run ID [int]
     - action [int: 0=walking (not in queue/exit), 1=queueing (start/not moving), 2=moving (moving in queue)]
     - bin-0 [float]
     - ...
     - bin-Z [float]
3. Classify/test
   - do Y measurements of accel over X time
   - apply filters of 'learn'-phase
   - for all records in 'classification' table, calculate difference (in Z-dimensional space)
   - Find k-nearest neighbours (weighted distance??), determine with majority voting the classification

**Comment by HesselM** [comment]
Optional: use BackPropagation (Neural network) for classification:
1. get accel data.  (just as before)
2. Filters/classification (just as before)
3. Learn
   - Neural network with:
     - Z-inputs (for each bin)
     - 3 outputs (for each action)
     - X hidden nodes (to be determined while testing)
4. Apply backpropagation
   - sigmoid as threshold function
   - weight as input (to be learned)
   - decaying learning rate (decay to be determined)
   - run on 70% of dataset sequentially, until each sample results in ~80% correctnes (normalise 3 outputs and correct output should be >0.8) 
5. test on 30% of dataset, if satisfactory, done! :D

**Comment by HesselM** [comment]
Update:

Currently, branch [activity_monitoring](https://github.com/HesselM/IN4254-SmartPhoneSensing/tree/activity_monitoring) contains the basic functions for the app. This includes
- linear accelerometer logging
- logging is annotated with 'walking', 'idle', 'queueing'.
- database for logging
- export to csv, which can be used in matlab
- matlab functions for parsing datalog
- calibration for constant-bias correction for accelerometer.

To use the app:
- put the app on the phone
- put the app on a table
- press 'calibrate'. It will calibrate for ~ 30 seconds
- bias-estimate is doen (Toast message)
- 'start accel' to start logging
- during log the type of activity may change

What to do:
- collect a lot of data!
- check the standard deviation of data with respect of the paper (matlab functions is already build)
- implement crosscorelation in matlab
- feature selection
  Maybe: different time-windows for deviation. Each window results in a single deviation. the group of deviations can be used as features. Hypothsis: idle should show a low magnitude on all windows. If a sudden movement is created, a spike is shown in the shorter windows, but will be neglected in the longer. Walking will result in large deviations in all windows. Queueig on the other hand is being idle, followed by a short burst of motion, hence a spike should be obseverd in the 'mid-sized' windows and lower. Based on these difference kNN might be able to classify the data

If it is not working:
- maybe change classification: step, idle and walking. Queueing is mostly being idle en taking a small step. Hence, a sequence of 'steps' can be determined to check the queue length and serving time.
